File: Team7/Team7/src/main/java/id/bca/co/team7/assetManagement/Team7/model/CheckIn.java
package id.bca.co.team7.assetManagement.Team7.model;

import javax.persistence.*;
import java.util.Date;

@Entity
@Table(name="checkin")
public class CheckIn {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private int id;

    @ManyToOne
    @JoinColumn(name = "asset_id")
    private Asset asset;

    @ManyToOne
    @JoinColumn(name = "warehouse_id")
    private Warehouse warehouse;

    @Column(name = "tanggal_masuk")
    private Date tanggal_masuk;

    @Column(name = "jumlah")
    private int jumlah;

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public Asset getAsset() {
        return asset;
    }

    public void setAsset(Asset asset) {
        this.asset = asset;
    }

    public Warehouse getWarehouse() {
        return warehouse;
    }

    public void setWarehouse(Warehouse warehouse) {
        this.warehouse = warehouse;
    }

    public Date getTanggal_masuk() {
        return tanggal_masuk;
    }

    public void setTanggal_masuk(Date tanggal_masuk) {
        this.tanggal_masuk = tanggal_masuk;
    }

    public int getJumlah() {
        return jumlah;
    }

    public void setJumlah(int jumlah) {
        this.jumlah = jumlah;
    }
}

File: team-7-front-end/src/app/model/checkin.ts
import {Asset} from "./asset";
import {Warehouse} from "./warehouse";

export interface Checkin {
  id: number,
  asset: Asset,
  warehouse: Warehouse,
  tanggal_masuk: string,
  jumlah : number
}

export interface Checkin1 {
  id: number,
  asset_id: number,
  warehouse_id: number,
  tanggal_masuk: string,
  jumlah : number
}

export class CheckinModel {
  constructor(
    public id: number,
    public asset_id: number,
    public warehouse_id: number,
    public tanggal_masuk : string,
    public jumlah : number
  ) { }
}
